### DISCUSSION TITLE
problem with configure self hosted keygen in docker compose

### DISCUSSION THREAD

**Discussion by ahnjs1987** [keygen-sh/community#155] [Issue]
Hello,

I want to configure self hosted keygen with docker compose.

I followed your instruction in the web page(https://keygen.sh/docs/self-hosting/)

after execute "docker compose up", I tried to reach to the server with ping api(https://{URL}/v1/ping)), it returns proper response. 

Then I want to generate token with api(/v1/token), it returns error.
{"meta":{"id":"a9e289d2-8ec8-4c82-ba66-4ea49648f193"},"errors":[{"title":"Internal server error","detail":"Looks like something went wrong! Our engineers have been notified. If you continue to have problems, please contact support@keygen.sh."}]}

How do I fix this?

**Comment by ezekg** [comment]
Look at the server logs. There will be a detailed error and stack trace.

**Comment by ahnjs1987** [comment]
I resolved the issue.

the issue is caused by env file setting(REDIS_URL).

I saw the logs from docker image named web, there was some log for redis authentication.
(because I want to use the separate redis vm for keygen, and the redis has ID/PW. but the I didn't wrote password on the env file.)

I wrote REDIS_URL=redis://:password@redis:6379 and re-run the docker compose up, It work properly.

Thanks for your effort!
